Source/Core/FrustumGeometry.js

```diff
@@ -0,0 +1,489 @@
+define([
+        './BoundingSphere',
+        './Cartesian3',
+        './Cartesian4',
+        './Check',
+        './ComponentDatatype',
+        './defaultValue',
+        './defined',
+        './Geometry',
+        './GeometryAttribute',
+        './GeometryAttributes',
+        './Matrix3',
+        './Matrix4',
+        './OrthographicFrustum',
+        './PerspectiveFrustum',
+        './PrimitiveType',
+        './Quaternion',
+        './VertexFormat'
+    ], function(
+        BoundingSphere,
+        Cartesian3,
+        Cartesian4,
+        Check,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        Geometry,
+        GeometryAttribute,
+        GeometryAttributes,
+        Matrix3,
+        Matrix4,
+        OrthographicFrustum,
+        PerspectiveFrustum,
+        PrimitiveType,
+        Quaternion,
+        VertexFormat) {
+    'use strict';
+
+    var PERSPECTIVE = 0;
+    var ORTHOGRAPHIC = 1;
+
+    /**
+     * Describes a frustum at the given position and orientation.
+     *
+     * @alias FrustumGeometry
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {PerspectiveFrustum|OrthographicFrustum} options.frustum The frustum.
+     * @param {Cartesian3} options.position The position of the frustum.
+     * @param {Quaternion} options.orientation The orientation of the frustum.
+     * @param {VertexFormat} [options.vertexFormat=VertexFormat.DEFAULT] The vertex attributes to be computed.
+     */
+    function FrustumGeometry(options) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('options', options);
+        Check.typeOf.object('options.frustum', options.frustum);
+        Check.typeOf.object('options.position', options.position);
+        Check.typeOf.object('options.orientation', options.orientation);
+        //>>includeEnd('debug');
+
+        var frustum = options.frustum;
+        var orientation = options.orientation;
+        var position = options.position;
+        var vertexFormat = defaultValue(options.vertexFormat, VertexFormat.DEFAULT);
+        var drawNearPlane = defaultValue(options._drawNearPlane, true);
```

Comment:
Add a comment about why this is private...or I wonder if we would want it to be public.